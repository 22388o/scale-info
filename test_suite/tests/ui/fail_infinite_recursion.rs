use scale_info::TypeInfo;
use scale::Encode;

#[derive(TypeInfo)]
struct Color<Hue>{hue: Hue}
#[derive(TypeInfo)]
struct Texture<Bump, Hump>{bump: Bump, hump: Hump}

#[allow(unused)]
#[derive(Encode, TypeInfo)]
struct Apple<T, U> {
    #[codec(compact)]
    one: Color<U>,   // <â€“ works with a "naked" generic, `U`, but not like this
    two: Texture<T, U>,
}

fn assert_type_info<T: TypeInfo + 'static>() {}

fn main() {
    // When this test fails it could mean that https://github.com/rust-lang/rust/issues/81785 is fixed
    assert_type_info::<Apple<u8, u16>>();
}
