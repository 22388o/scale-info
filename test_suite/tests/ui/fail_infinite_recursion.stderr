error[E0275]: overflow evaluating the requirement `_parity_scale_codec::Compact<_>: Decode`
  |
  = help: consider adding a `#![recursion_limit="256"]` attribute to your crate (`$CRATE`)
  = note: required because of the requirements on the impl of `Decode` for `_parity_scale_codec::Compact<_>`
  = note: 126 redundant requirements hidden
  = note: required because of the requirements on the impl of `Decode` for `_parity_scale_codec::Compact<_>`
